packages/circuits/src/email_auth_template.circom

```diff
@@ -182,6 +182,176 @@ template EmailAuth(n, k, max_header_bytes, max_subject_bytes, recipient_enabled)
+    var command_field_len = compute_ints_size(max_command_bytes);
+    var domain_len = domain_len_const();
+    var domain_field_len = compute_ints_size(domain_len);
+    var k2_chunked_size = k >> 1;
+    if(k % 2 == 1) {
+        k2_chunked_size += 1;
+    }
+    var timestamp_len = timestamp_len_const();
+    var code_len = invitation_code_len_const();
+
+
+    signal output domain_name[domain_field_len];
+    signal output public_key_hash;
+    signal output email_nullifier;
+    signal output timestamp;
+    signal output masked_command[command_field_len];
+    signal output account_salt;
+    signal output is_code_exist;
+    
+    // Verify Email Signature
+    component email_verifier = EmailVerifier(max_header_bytes, max_body_bytes, n, k, 0, 0, 0);
+    email_verifier.emailHeader <== padded_header;
+    email_verifier.emailHeaderLength <== padded_header_len;
+    email_verifier.pubkey <== public_key;
+    email_verifier.signature <== signature;
+    email_verifier.bodyHashIndex <== body_hash_idx;
+    email_verifier.precomputedSHA <== precomputed_sha;
+    email_verifier.emailBody <== padded_body;
+    email_verifier.emailBodyLength <== padded_body_len;
+    public_key_hash <== email_verifier.pubkeyHash;
+
+    // FROM HEADER REGEX
+    signal from_regex_out, from_regex_reveal[max_header_bytes];
+    (from_regex_out, from_regex_reveal) <== FromAddrRegex(max_header_bytes)(padded_header);
+    from_regex_out === 1;
+    signal from_email_addr[email_max_bytes];
+    from_email_addr <== SelectRegexReveal(max_header_bytes, email_max_bytes)(from_regex_reveal, from_addr_idx);
+
+    // DOMAIN NAME HEADER REGEX
+    signal domain_regex_out, domain_regex_reveal[email_max_bytes];
+    (domain_regex_out, domain_regex_reveal) <== EmailDomainRegex(email_max_bytes)(from_email_addr);
+    domain_regex_out === 1;
+    signal domain_name_bytes[domain_len];
+    domain_name_bytes <== SelectRegexReveal(email_max_bytes, domain_len)(domain_regex_reveal, domain_idx);
+    domain_name <== Bytes2Ints(domain_len)(domain_name_bytes);
+    
+    signal sign_hash;
+    signal sign_ints[k2_chunked_size];
+    (sign_hash, sign_ints) <== HashSign(n,k)(signature);
+    email_nullifier <== EmailNullifier()(sign_hash);
+
+    // Timestamp regex + convert to decimal format
+    signal timestamp_regex_out, timestamp_regex_reveal[max_header_bytes];
+    (timestamp_regex_out, timestamp_regex_reveal) <== TimestampRegex(max_header_bytes)(padded_header);
+    signal timestamp_str[timestamp_len];
+    timestamp_str <== SelectRegexReveal(max_header_bytes, timestamp_len)(timestamp_regex_reveal, timestamp_idx);
+    signal raw_timestamp <== Digit2Int(timestamp_len)(timestamp_str);
+    timestamp <== timestamp_regex_out * raw_timestamp;
+
+    // Extract the command from the body
+    signal command_reveal[max_command_bytes];
+    component select_command_sub_array = SelectSubArray(max_body_bytes, max_command_bytes)(padded_body, command_idx, command_len);
```

Comment:
Where is a definition of `SelectSubArray`?
I expect that it extracts a substring from `command_idx` to `command_idx + command_len`, but it has an attack vector because a relayer can choose any substring in the email body.
To prevent this, the circuit first needs to do a regex validation of `padded_body` for the regex `<div id="zkemail">[^<>\r\n/]</div>` and gets a revealed string for `[^<>\r\n/]` as `command_reveal`.

Reply:
It implies that an input `command_len` is unnecessary.